src/store/actions.ts

```diff
@@ -436,6 +444,69 @@ export const submit: any = async ({ commit, getters, dispatch }): Promise<any> =
   }
 }
 
+/**
+ * Re-submits an expired NR (without changing the current NR data - in case of failure).
+ **/
+export const resubmit: any = async ({ commit, getters }): Promise<boolean> => {
+  // safety check
+  if (getters.getEditMode) {
+    console.error('resubmit() - should not be edit mode') // eslint-disable-line no-console
+    return false
+  }
+
+  const nrData = getters.getNr
+
+  // safety check
+  if (!nrData) {
+    console.error('resubmit() - could not get current NR data') // eslint-disable-line no-console
+    return false
+  }
+
+  // commit the original NR's data
+  commitExistingData({ commit, getters })
+
+  // build the request data
+  const nrTypeData = getNrTypeData({ getters }, NrType.DRAFT)
+
+  // override resubmission message in Additional Info field
+  nrTypeData['additionalInfo'] = '*** Resubmission of ' + nrData['nrNum'] + ' ***'
+
+  // override names data
+  nrTypeData['names'] = nrData.names.map(name => ({
+    choice: name.choice,
+    consent_words: '',
+    conflict1: '',
+    conflict1_num: '',
+    designation: getNameDesignation(name),
+    name: name.name,
+    name_type_cd: name.name_type_cd
+  }))
+
+  // override request action code
+  nrTypeData['request_action_cd'] = RequestCode.RESUBMIT
+
+  // post new NR (without adding comment)
+  const request = await NamexServices.postNameRequests(getters.getRequestActionCd, nrTypeData, false)
+  if (request) {
+    commit('setNrResponse', request)
+    return true
+  }
+
+  return false
+}
+
+/**
+ * Sometimes name objects have a null designation (because the designation is part of the name).
```

Comment:
This is probably an unintended consequence of a previous change, @cameron-freshworks , should this be fixed?

Reply:
I'm not too sure, i don't recall implementing this, I most likely copy pasted this over.
Not too sure of the implications of this.

Reply:
OK, I won't worry about it.